qbertclient/qbert.py

```diff
@@ -296,15 +296,24 @@ def get_master_ip(self, cluster_uuid):
         LOG.debug('Getting masterIp for cluster %s', cluster_uuid)
         return self.get_cluster(cluster_uuid)['masterIp']
 
-    def get_kubeconfig(self, cluster_uuid):
+    def get_kubeconfig(self, cluster_uuid, username='', password=''):
         """
-        Get kubeconfig of a cluster by uuid
+        Get kubeconfig of a cluster by uuid. If both username and password
+        are supplied, then configure kubeconfig to use password-based
+        authentication, else use token.
         :param cluster_uuid:
+        :param username: optional username
+        :param password: optional password
         :return:
         """
         endpoint = '/kubeconfig/{0}'.format(cluster_uuid)
         resp = self._make_req(endpoint)
-        kubeconfig = resp.text.replace('__INSERT_BEARER_TOKEN_HERE__', self.token)
+        if username and password:
+            s = '{"username":"%s","password":"%s"}' % (username, password)
+            token = base64.b64encode(s.encode()).decode()
+        else:
+            token = self.token
```

Comment:
Should this be a new token valid for a new 24 hours?

Reply:
I did not change the existing behavior for tokens: the token created when the qbert client object was created is used for the kubeconfig. We could refresh it, but that would be a separate change.